ygen/codegen.go

```diff
@@ -443,7 +443,9 @@ func (cg *YANGCodeGenerator) GenerateGoCode(yangFiles, includePaths []string) (*
 	}
 
 	// Store the returned schematree and enumSet within the state for this code generation.
-	gogen := newGoLangMapper(mdef.schematree, enumSet)
+	gogen := NewGoLangMapper(cg.Config.GoOptions.GenerateSimpleUnions)
```

Comment:
Should we deprecate the more complex unions at this point?

Reply:
I'm open to doing that; although I'm wondering whether we should keep the backwards-incompatible changes all concentrated in a single release at v1 to minimize the number of times downstream users have to change?